pkg/execution/state/redis_state/redis_state.go

```diff
@@ -339,34 +332,45 @@ func (m mgr) UpdateMetadata(ctx context.Context, runID ulid.ULID, md state.Metad
 	return nil
 }
 
-func (m mgr) IsComplete(ctx context.Context, runID ulid.ULID) (bool, error) {
-	cmd := m.r.B().Hget().Key(m.kf.RunMetadata(ctx, runID)).Field("status").Build()
-	val, err := m.r.Do(ctx, cmd).AsBytes()
+func (m shardedMgr) IsComplete(ctx context.Context, runID ulid.ULID) (bool, error) {
+	fnRunState := m.s.FunctionRunState()
+	r := fnRunState.Client(runID)
+	val, err := r.Do(ctx, func(client rueidis.Client) rueidis.Completed {
+		return client.B().Hget().Key(fnRunState.kg.RunMetadata(ctx, runID)).Field("status").Build()
+	}).AsBytes()
 	if err != nil {
 		return false, err
 	}
 	return !bytes.Equal(val, []byte("0")), nil
 }
 
-func (m mgr) Exists(ctx context.Context, runID ulid.ULID) (bool, error) {
-	cmd := m.r.B().Exists().Key(m.kf.RunMetadata(ctx, runID)).Build()
-	return m.r.Do(ctx, cmd).AsBool()
+func (m shardedMgr) Exists(ctx context.Context, runID ulid.ULID) (bool, error) {
+	fnRunState := m.s.FunctionRunState()
+	r := fnRunState.Client(runID)
+	return r.Do(ctx, func(client rueidis.Client) rueidis.Completed {
```

Comment:
Unfortunately, we needed to update all occurrences of Do() calls with the new retriable syntax. The actual commands did not change other than using the new key generators which may return sharded keys if the Run ID is newer than the switchover date.